### DISCUSSION TITLE
Using the verbal-free-recall.js plugin

### DISCUSSION THREAD

**Discussion by radrogue1** [jspsych/jsPsych#901] [Q&A] [answered]
Hi all,

I am currently writing a verbal free recall experiment. I was hoping to use the verbal-free-recall.js plugin in order to record the participants' responses, but cannot seem to get the plug-in to run correctly. I have a random image file as the stimulus, in an attempt to get the plugin to run.

Here is my code for this section thus far: 

```
var recordResponses = {
        type: "audio-response",
        stimulus: "img/blue.png"
    };
    
    timeline.push(recordResponses);
    
    jsPsych.init({
        timeline: timeline
    });

```
When I run the code I have, the preloading progress bar comes up and the code does not advance past that point.

Thanks in advance for the help!

**Comment by vijaymarupudi** [comment, marked as answer]
Can you check for errors in your browser console?

Firefox: https://developer.mozilla.org/en-US/docs/Tools/Web_Console/Opening_the_Web_Console\
Chrome: https://developers.google.com/web/tools/chrome-devtools/open#console

Additionally, I don't think `audio-response` is a default jsPsych plugin. Could you elaborate on that?

**Reply by radrogue1** [reply]
I'm using Safari but I can start running my experiment in Chrome if necessary.
 
Link to Plugin: https://github.com/kahana-sysadmin/jsPsych/blob/master/plugins/verbal-free-recall.js

On the plugin, it appears to use two names, "image-audio-response" and "audio-response". However, attempting to use either name does not work.

I do not know if it is a default plugin, but I did have to manually download it, as it did not download with the rest of the files in the jsPsych download.

**Reply by vijaymarupudi** [reply]
There are instructions for Safari at [this link](https://appuals.com/open-browser-console/).

That plugin file has an error in it, remove the last line, which says:

```
© 2020 GitHub, Inc.
```

You should use it as `image-audio-response`. If it doesn't work, post console error output please :)

**Reply by radrogue1** [reply]
Now that I have gotten the plugin to run, it is throwing me a new error that I have included below. If you can think of anything to help I would appreciate it.
<img width="1440" alt="Screen Shot 2020-07-21 at 9 55 40 PM" src="https://user-images.githubusercontent.com/68128449/88125279-f12fcd80-cb9c-11ea-90d3-40fa35679043.png">

**Reply by vijaymarupudi** [reply]
I would recommend using another browser (Safari is truly the new IE, it's a horror to work with and I always recommend people to not use it). This might mean that you will have to exclude participants who use Safari. I would recommend using Firefox.

As you can see [here](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder/MediaRecorder#Browser_compatibility), Safari doesn't have the MediaRecorder API yet.

**Reply by vijaymarupudi** [reply]
You could use [this](https://github.com/ai/audio-recorder-polyfill), but that's more complicated.

**Reply by radrogue1** [reply]
Ok, I've switched over to Chrome and it appears to be recording audio when I run the function. The only issue is that I cannot seem to access the recording that I make using that function. I've looked at all of the jsPsych.data stuff and in the actual plugin itself. Is there a way to output a .wav or .mp3 from this plugin?

Thank you once again for all your help, you've been a huge help.

**Reply by vijaymarupudi** [reply]
I'm not that familiar with audio data, so I can't say this for sure, but the answer seems to be [here], on this line

[here]: https://stackoverflow.com/questions/52021331/convert-blob-to-wav-file-without-loosing-data-or-compressing

```javascript
const blobDataInWavFormat: Blob = new Blob([trial.audio_data], { type : 'audio/wav;' })
```

You'll need to access the `audio_data` field in the plugin data. Consult the page on how to download it.

**Comment by radrogue1** [comment]
That worked! Thank you so much for your help!
